aten/src/ATen/core/ivalue.cpp

```diff
@@ -249,6 +249,41 @@ void IValue::dump() const {
   std::cout << *this << "\n";
 }
 
+IValue IValue::deepcopy() const {
+  switch(tag) {
+    case IValue::Tag::Tensor:
+      return IValue(toTensor().clone());
+    case IValue::Tag::Tuple: {
+      std::vector<IValue> copied_elements;
+      for (const auto& e : toTuple()->elements()) {
+        copied_elements.push_back(e.deepcopy());
+      }
+      return IValue(ivalue::Tuple::create(copied_elements));
+    }
+    case IValue::Tag::String:
+      return IValue(ivalue::ConstantString::create(toString()->string()));
+    case IValue::Tag::GenericList:
+      return IValue(toList().copy());
+    case IValue::Tag::GenericDict:
+      return IValue(toGenericDict().copy());
+    case IValue::Tag::Object: {
+      return IValue(toObject()->deepcopy());
+    case IValue::Tag::None:
+    case IValue::Tag::Double:
+    case IValue::Tag::Int:
+    case IValue::Tag::Bool:
+    case IValue::Tag::Device:
+    case IValue::Tag::Uninitialized:
+    case IValue::Tag::Blob:
+    case IValue::Tag::Capsule:
+    case IValue::Tag::Future:
+    case IValue::Tag::PyObject:
```

Comment:
Blob, Capsule, Future, and PyObject are all potentially mutable, so deepcopy is not correctly defined.

Reply:
Can we deepcopy those?